pkg/syncer/status/suite_test.go

```diff
@@ -0,0 +1,268 @@
+				},
+				Spec: gatewayv1beta1.GatewaySpec{
+					GatewayClassName: "test",
+					Listeners: []gatewayv1beta1.Listener{
+						{
+							Name:     "listener-1",
+							Hostname: &hostname,
+							Port:     443,
+							Protocol: gatewayv1beta1.HTTPSProtocolType,
+						},
+					},
+				},
+			}
+		})
+
+		//ensure gateway is removed after each test.
+		AfterEach(func() {
+			// Delete gateway from control plane
+			Expect(func() bool {
+				err := k8sClient.Delete(ctx, gateway)
+				return err == nil || apierrors.IsNotFound(err)
+			}()).To(BeTrue())
+
+			// Confirm it's removed downstream
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: gateway.Name, Namespace: dataPlaneNS}, &gatewayv1beta1.Gateway{})
+				return apierrors.IsNotFound(err)
+			}, TestTimeoutMedium, TestRetryIntervalMedium).Should(BeTrue())
+		})
+
+		It("should annotate the status to the upstream resource", func() {
+			metadata.AddAnnotation(gateway, syncer.MCTC_SYNC_ANNOTATION_PREFIX+clusterID, "true")
+
+			//we will be creating the data plane gateway (as spec syncer is not running)
+			dataplaneGateway := gateway.DeepCopy()
+			dataplaneGateway.Namespace = dataPlaneNS
+
+			//create control plane gateway
+			Expect(k8sClient.Create(ctx, gateway)).To(BeNil())
+			controlPlaneGateway := &gatewayv1beta1.Gateway{}
+			gatewayNamespacedName := types.NamespacedName{Name: gateway.Name, Namespace: controlPlaneNS}
+
+			//create data plane gateway with status
+			Expect(k8sClient.Create(ctx, dataplaneGateway)).To(BeNil())
+
+			dataplaneGateway.Status.Listeners = []gatewayv1beta1.ListenerStatus{
+				{
+					Name: "test-status-listener",
+					SupportedKinds: []gatewayv1beta1.RouteGroupKind{
+						{
+							Kind: "ingress",
+						},
+					},
+					Conditions: []metav1.Condition{
+						{
+							Type:               "testing",
+							Status:             "True",
+							ObservedGeneration: 1,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "testing",
+							Message:            "testing",
+						},
+					},
+				},
+			}
+
+			ctrl.Log.Info("updating data plane gateway status")
+			Expect(k8sClient.Status().Update(ctx, dataplaneGateway.DeepCopy())).To(BeNil())
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: dataplaneGateway.Name, Namespace: dataplaneGateway.Namespace}, dataplaneGateway)
+				if err != nil {
+					ctrl.Log.Error(err, "error getting data plane gateway")
+					return false
+				}
+				//ctrl.Log.Info("got data plane gateway", "status", dataplaneGateway.Status)
+
+				err = k8sClient.Get(ctx, gatewayNamespacedName, controlPlaneGateway)
+				if err != nil {
+					ctrl.Log.Error(err, "error getting control plane gateway")
+					return false
+				}
+				//ctrl.Log.Info("got control plane gateway", "annotations", controlPlaneGateway.Annotations)
+				return metadata.HasAnnotation(controlPlaneGateway, SyncerClusterStatusAnnotationPrefix+clusterID)
```

Comment:
Would it be better to verify the annotation contents here as well?

Reply:
I considered that alright, but wasn't sure if it'd get a bit noisy. I'll add it.

Reply:
I've added this now, deepequals was being a little weird with me, but I've got something that works.